### ISSUE TITLE
New URLs for listing pickups

### ISSUE THREAD

**Issue by matheusruschel** [VanHack/binners-project-api#48]
We need two URLs, one to fetch the "completed" pickups and the other one to fetch the "on going" and "Waiting for Review" pickups

**Comment by emersonmellado** [comment]
We also need a way to filter this listings. Let's say to do the easy way: 

pickups?skip=10&list10 could bring the 10-20 list and so on.

**Comment by matheusruschel** [comment]
Yes, that would be perfect, I already have some code that manipulates that functionality in iOS

**Comment by matheusruschel** [comment]
Sooooooooooo...Is it up already? =)

**Comment by soeirosantos** [comment]
Hey, guys, I'm on it, but I have some doubts:

On the model there is no field to hold these "completed", "on going" and "waiting for review" statuses.
Will we create this field? If so, who will change this status, in a business point of view?

I also saw that there is no review on the model... what  "waiting for review" means?

Can you clarify these points for me? 

Thanks!

**Comment by emersonmellado** [comment]
Good question, I guess we can keep the field with the default value of "on going ".

This one was created to separate the pickups, all should come with "waiting for review " then a staff will change it to "om going ".

**Comment by matheusruschel** [comment]
Hey great to see you here!! Let me clear out some stuff.

1- yes, those fields have to be created for the pickup, we don't have them yet.

2- the status will be changed by the user's actions. The pickup starts as "on going", it is changed to "waiting for review" after the binner says it has completed the pickup. Finally the status changes to "Completed" after the user reviews how was his experience for the pickup.

3 - waiting for review means that the user has to review the pickup for the status to be changed to completed. There should be a rating structure inside the pickup, it should contain a string for comments and a 1-10 score of how good was the pickup. 

Hope I could help! If you have any other questions just ask away!

**Comment by soeirosantos** [comment]
@matheusruschel ,

Just for check:

> Finally the status changes to "Completed" after the user reviews how was his experience for the pickup.

Only the requester can review the pickup, right? I'm going to include this check when a review is sent, ok?

**Comment by soeirosantos** [comment]
Guys, I've done this task. 

Just a brief explanation:

`PATCH /api/v1.0/pickups/{_id}/done` 
Puts a pickup as done, status is set to 'waiting review'

`POST /api/v1.0/pickups/{_id}/review` 
Place a review for a pickup and status is set to 'completed'

`GET /api/v1.0/pickups/completed`
List last 6 months user's pickups completed

`GET /api/v1.0/pickups/ongoing`
List last 6 months user's pickups on going

`GET /api/v1.0/pickups/waitingreview`
List last 6 months user's pickups waiting for review

If we need some change, please, let me know.

**Comment by matheusruschel** [comment]
Yes, you're correct about the requester being the only one able to review the pickup.

Awesome work man!! In this rhythm we will be over pretty soon!

**Comment by matheusruschel** [comment]
Hey we need another url, could you build them please?

We need something to fetch ongoing pickups and wating for review pickups at the same time since they're gonna be displayed at the same page, can we get them ordered by time(most recent to oldest)?

`GET /api/v1.0/pickups/waitingreview=30&ongoing=30`

Can we make something like that?

Thanks you rock =)

**Comment by matheusruschel** [comment]
Forgot something. We also need a url to Cancel the pickup, and 3 other urls

-  change the pickup time
-  change the pickup day
- change the pickup address (includes coordinates and address)

Thanks!

**Comment by soeirosantos** [comment]
> We need something to fetch ongoing pickups and wating for review pickups at the same time since they're gonna be displayed at the same page, can we get them ordered by time(most recent to oldest)?

@matheusruschel can you perform two requests, like below, and order the results on the client? 

`GET /api/v1.0/pickups/waitingreview?limit=30`
`GET /api/v1.0/pickups/ongoing?limit=30`

This approach let the API a little bit more generic and plain, delegating to client perform specific view operations. Of course, if you say that you really need that another approach I can do it.

**Comment by soeirosantos** [comment]
> Forgot something. We also need a url to Cancel the pickup, and 3 other urls

Please, use:

`PATCH /api/v1.0/pickups/{_id}/canceled` to change the status for canceled.

> change the pickup time
> change the pickup day
> change the pickup address (includes coordinates and address)

And, use:

`PUT /api/v1.0/pickups/{_id}` to update the pickup. Observe that the model schema (refer to API doc) only allow changing the fields that you asked, ok?

I'm going to submit the PR.

Thanks by let me participate and help!! :)

**Comment by matheusruschel** [comment]
Ok! Thanks @soeirosantos, you rock!

**Comment by soeirosantos** [comment]
@matheusruschel, I've created the below endpoint.

Example: 

`/api/v1.0/pickups/status-tracking?status=ongoing&status=canceled&limit=30`

Values allowed for status: `ongoing`, `waitingreview`, `completed` and `canceled`.

I think this endpoint deprecates the specifics endpoints for each status. If so, we can remove them in future.

Please, referer the API doc for details.B

**Comment by matheusruschel** [comment]
Thank you man! Yes it does, sorry for wasting your time. Anyway, we're close to the end now I guess
